File: Code/new_fitervari_backend/src/main/java/at/htl/control/WorkoutRepository.java
package at.htl.control;

import at.htl.model.Workout;
import io.quarkus.hibernate.orm.panache.PanacheRepository;

import javax.enterprise.context.ApplicationScoped;

@ApplicationScoped
public class WorkoutRepository implements PanacheRepository<Workout> {

    public Workout save(Workout entity) {
        return getEntityManager().merge(entity);
    }
}

File: Code/fitervari_backend/src/main/java/at/htl_leonding/repository/WorkoutRepository.kt
package at.htl_leonding.repository

import at.htl_leonding.model.Workout
import at.htl_leonding.model.WorkoutHistory
import io.quarkus.hibernate.orm.panache.PanacheRepository
import javax.enterprise.context.ApplicationScoped
import javax.inject.Inject
import javax.persistence.EntityManager

@ApplicationScoped
class WorkoutRepository : PanacheRepository<Workout> {
    @Inject
    lateinit var em: EntityManager

    fun save(workout: Workout): Workout {
        return em.merge(workout)
    }
}